### PR TITLE
support --suppressions-list in cppcheck

### PR DESCRIPTION
**Pull request by unhammer**
Some projects use a file for this, would be handy.

### CODE REVIEW

File: flycheck.el [hunk 1]
```diff
@@ -6556,6 +6556,7 @@ See URL `http://cppcheck.sourceforge.net/'."
             (option-list "-I" flycheck-cppcheck-include-path)
             (option-list "--std=" flycheck-cppcheck-standards concat)
             (option-list "--suppress=" flycheck-cppcheck-suppressions concat)
+            (option "--suppressions-list=" flycheck-cppcheck-suppressions-file concat)
```
Comment (fmdkdd) [inline flycheck.el]: Do we have a `flycheck-cppcheck-suppressions-file` variable?  I'm not seeing any.  Am I missing something, or do you just `setq` the variable in your config?

Comment (unhammer) [reply flycheck.el]: Oh, I had it in dir-locals.el so I didn't notice it missing; added in 5ce24713c8f00b42fa3286c10b5951fe4f114a39

File: flycheck.el [hunk 2]
```diff
@@ -6511,6 +6511,18 @@ non-nil, pass the standards via one or more `--std=' options."
...
+(flycheck-def-option-var flycheck-cppcheck-suppressions-file nil c/c++-cppcheck
+  "The suppressions file to use in cppcheck.
+
+The value of this variable is a file with the suppressions to
+use, or nil to pass nothing to cppcheck.  When non-nil, pass the
+suppressions file via the `--suppressions-list=' option."
+  :type '(choice (const :tag "Default" nil)
+                 (file :tag "Suppressions file"))
+  :safe #'stringp
+  :package-version '(flycheck . "31"))
```
Comment (fmdkdd) [inline flycheck.el]: 31 is already out, so that variable would go into version 32.

**Comment by unhammer** [comment]
Should this be rebased/squashed, or is it OK like this?

**Comment by cpitclaudel** [comment]
I like clean commits; a rebase would be nice :)

**Comment by unhammer** [comment]
rebased

**Comment by fmdkdd** [comment]
Thank you! 👍

**Pull request merged**